### PR TITLE
Dyn 5530 library zoom scaling

### PR DESCRIPTION
**Pull request by filipeotero**
### Purpose

The purpose of this PR is to add a zoom scale for the Library in the prefence settings. 

zoomScaleLibrary

### Declarations

Check these if you believe they are true

- [ ] The codebase is in a better state after this PR
- [ ] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.
- [ ] This PR modifies some build requirements and the readme is updated

### Reviewers

@QilongTang @zeusongit 

### FYIs

@RobertGlobant20

### CODE REVIEW

File: src/DynamoCoreWpf/Views/Core/DynamoView.xaml.cs [hunk 1]
```diff
@@ -98,6 +99,12 @@ public partial class DynamoView : Window, IDisposable
 
         internal ShortcutToolbar ShortcutBar { get { return shortcutBar; } }
 
+        internal PreferencesView PreferencesWindow {
+            get { return preferencesWindow; }
+        }
+
+        internal event Action OnPreferencesWindowChanged;
```
Comment (filipeotero) [inline src/DynamoCoreWpf/Views/Core/DynamoView.xaml.cs:106]: The library extension didn't have access to the new PreferencesView object created. So, this event was added to be called after a new preferences window is opened.

File: src/LibraryViewExtensionWebView2/LibraryViewController.cs [hunk 2]
```diff
@@ -81,6 +88,16 @@ internal LibraryViewController(Window dynamoView, ICommandExecutive commandExecu
...
         }
 
+        private void Browser_ZoomFactorChanged(object sender, EventArgs e)
+        {
+            dynamoViewModel.Model.PreferenceSettings.LibraryZoomScale = browser.ZoomFactor;
+        }
+
+        void PreferencesWindowChanged()
+        {
+            this.dynamoView.PreferencesWindow.LibraryZoomScalingSlider.ValueChanged += DynamoSliderValueChanged;
```
Comment (filipeotero) [inline src/LibraryViewExtensionWebView2/LibraryViewController.cs:98]: When a new PreferencesWindow is created, the value of the slider is subscribed.

File: src/DynamoCoreWpf/Views/Menu/PreferencesView.xaml [hunk 3]
```diff
@@ -1258,9 +1259,70 @@
...
+                                            <TextBlock Text="100%" Grid.Column="1"></TextBlock>
+                                            <TextBlock Text="300%" Grid.Column="2"></TextBlock>
+                                        </Grid>
+                                    </StackPanel>
+                                </Grid>
+                            </Expander>
                         </Grid>
                     </TabItem>
-
+
```
Comment (zeusongit) [inline src/DynamoCoreWpf/Views/Menu/PreferencesView.xaml:1325]: extra spaces

File: src/DynamoCoreWpf/Views/Core/DynamoView.xaml.cs [hunk 4]
```diff
@@ -1866,9 +1873,10 @@ private void OnDebugModesClick(object sender, RoutedEventArgs e)
 
         private void OnPreferencesWindowClick(object sender, RoutedEventArgs e)
         {
-            var preferencesWindow = new PreferencesView(this);
+            preferencesWindow = new PreferencesView(this);
+            OnPreferencesWindowChanged();
             preferencesWindow.WindowStartupLocation = WindowStartupLocation.CenterOwner;
-            preferencesWindow.ShowDialog();
+            preferencesWindow.ShowDialog();
```
Comment (zeusongit) [inline src/DynamoCoreWpf/Views/Core/DynamoView.xaml.cs:1879]: extra spaces

**Comment by zeusongit** [comment]
Can we have a pointer at 200% as well? I think it will look more natural that way, what do you think @Amoursol @Jingyi-Wen

**Comment by Amoursol** [comment]
> Can we have a pointer at 200% as well? I think it will look more natural that way, what do you think @Amoursol @Jingyi-Wen

Yes please - a consistent spread across would be best.

**Comment by filipeotero** [comment]
Added 200% pointer
image

File: test/DynamoCoreWpfTests/PackageManagerViewExtensionTests.cs [hunk 5]
```diff
@@ -242,7 +242,7 @@ public void PackageManagerViewExtesion_TriesToLoadLayoutSpecForBuiltInPackages()
         [Test]
         public void PackageManagerViewExtesion_SendsNotificationForPackagesThatTargetDifferentHost_AtExtensionLoad()
         {
-            var count = 0;
+            int count = 0;
```
Comment (QilongTang) [inline test/DynamoCoreWpfTests/PackageManagerViewExtensionTests.cs:245]: Is this change must have?

File: src/DynamoCoreWpf/Views/Menu/PreferencesView.xaml.cs [hunk 6]
```diff
@@ -27,6 +28,7 @@ public partial class PreferencesView : Window
     {
         private readonly PreferencesViewModel viewModel;
         private readonly DynamoViewModel dynViewModel;
+        private readonly DynamoView dynView;
```
[the remark this answers is not in the document]
Comment (filipeotero) [reply src/DynamoCoreWpf/Views/Menu/PreferencesView.xaml.cs]: There is no reference for this variable anymore. Removed. Thanks!

File: test/DynamoCoreWpfTests/PackageManagerViewExtensionTests.cs (hunk 5 above)
Comment (filipeotero) [reply test/DynamoCoreWpfTests/PackageManagerViewExtensionTests.cs]: Nope. Undoing this change... Thanks

File: test/settings/DynamoSettings-NewSettings.xml [hunk 7]
```diff
@@ -8,6 +8,7 @@
   <ShowCodeBlockLineNumber>false</ShowCodeBlockLineNumber>
   <ShowConnector>false</ShowConnector>
   <ShowConnectorToolTip>false</ShowConnectorToolTip>
+  <LibraryZoomScale>2.0361751152073739</LibraryZoomScale>
```
Comment (QilongTang) [inline test/settings/DynamoSettings-NewSettings.xml:11]: Is the value intended to be so long? Can we limit the digit length of the value serialized somehow?

File: src/LibraryViewExtensionWebView2/LibraryViewController.cs [hunk 8]
```diff
@@ -42,6 +44,8 @@ public class LibraryViewController : IDisposable
         // TODO remove this when we can control the library state from Dynamo more precisely.
         private bool disableObserver = false;
 
+        private static readonly string LibrarSlider = "notificationsButton";
```
Comment (QilongTang) [inline src/LibraryViewExtensionWebView2/LibraryViewController.cs:47]: Where is this used?

**Review by QilongTang — COMMENTED** [review]
LGTM with a couple comments, would you address those?

File: test/settings/DynamoSettings-NewSettings.xml (hunk 7 above)
Comment (filipeotero) [reply test/settings/DynamoSettings-NewSettings.xml:11]: We can use float instead of double

Comment (QilongTang) [reply test/settings/DynamoSettings-NewSettings.xml:11]: Works for me

**Comment by filipeotero** [comment]
I've also added localization for the expander header and adjusted the position of the percentage labels depending on the slider value.

**Review by QilongTang — APPROVED** [review]

**Comment by QilongTang** [comment]
@RobertGlobant20 With your latest changes, the test [WorkspaceContextMenu_TestIfInCanvasSearchHidesOnOpeningContextMenu](https://master-5.jenkins.autodesk.com/job/Dynamo/job/DynamoSelfServe/job/pullRequestValidation/8542/testReport/junit/DynamoCoreWpfTests/CoreUserInterfaceTests/WorkspaceContextMenu_TestIfInCanvasSearchHidesOnOpeningContextMenu/) still fails sporadically.

**Pull request merged**